inventory/sample/group_vars/all/docker.yml

```diff
@@ -15,10 +15,12 @@ docker_daemon_graph: "/var/lib/docker"
 
 ## A string of extra options to pass to the docker daemon.
 ## This string should be exactly as you wish it to appear.
-## An obvious use case is allowing insecure-registry access
-## to self hosted registries like so:
+## An obvious use case is allowing insecure-registry access to self hosted registries.
+## Can be ipddress and domain_name.
+## example define http://172.19.16.11 or http://mirror.registry.io
+#insecure_registry: http://mirror.registry.io
 docker_options: >-
-  --insecure-registry={{ kube_service_addresses }} --graph={{ docker_daemon_graph }} {{ docker_log_opts }}
+  --insecure-registry={{ insecure_registry | default(kube_service_addresses) }} --graph={{ docker_daemon_graph }} {{ docker_log_opts }}
```

Comment:
I think defining an external insecure registry may not prevent from creating one in the cluster (in addition to the one in the addons which gets accessed to through a proxy).
Imho, `insecure_registry` should be a list containing "{{ kube_service_addresses }}" by default (and maybe be renamed to `insecure_registries`).
In this case, this part should be modified as follow:
```
{{ insecure_registries | map('regex_replace', '^(.*)$', '--insecure-registry=\\1' ) | list | join(' ') }}
```